x-pack/plugins/infra/server/routes/log_analysis/results/log_entry_categories.ts

```diff
@@ -5,36 +5,29 @@
  */
 
 import Boom from 'boom';
-
-import { pipe } from 'fp-ts/lib/pipeable';
-import { fold } from 'fp-ts/lib/Either';
-import { identity } from 'fp-ts/lib/function';
-import { schema } from '@kbn/config-schema';
-import { InfraBackendLibs } from '../../../lib/infra_types';
 import {
-  LOG_ANALYSIS_GET_LOG_ENTRY_CATEGORIES_PATH,
   getLogEntryCategoriesRequestPayloadRT,
   getLogEntryCategoriesSuccessReponsePayloadRT,
+  LOG_ANALYSIS_GET_LOG_ENTRY_CATEGORIES_PATH,
 } from '../../../../common/http_api/log_analysis';
-import { throwErrors } from '../../../../common/runtime_types';
-import { NoLogAnalysisResultsIndexError } from '../../../lib/log_analysis';
-
-const anyObject = schema.object({}, { unknowns: 'allow' });
+import { createValidationFunction } from '../../../../common/runtime_types';
+import type { InfraBackendLibs } from '../../../lib/infra_types';
+import {
+  getTopLogEntryCategories,
+  NoLogAnalysisResultsIndexError,
+} from '../../../lib/log_analysis';
+import { assertHasInfraMlPlugins } from '../../../utils/request_context';
 
-export const initGetLogEntryCategoriesRoute = ({
-  framework,
-  logEntryCategoriesAnalysis,
-}: InfraBackendLibs) => {
+export const initGetLogEntryCategoriesRoute = ({ framework }: InfraBackendLibs) => {
   framework.registerRoute(
     {
       method: 'post',
       path: LOG_ANALYSIS_GET_LOG_ENTRY_CATEGORIES_PATH,
       validate: {
-        // short-circuit forced @kbn/config-schema validation so we can do io-ts validation
-        body: anyObject,
+        body: createValidationFunction(getLogEntryCategoriesRequestPayloadRT),
```

Comment:
How do we call this function without having to provide the generic values it specifies (`<DecodedValue, EncodedValue, InputValue>`)?

Reply:
I'm not sure I understand. The generic arguments are inferred from the codec passed as the function argument.

Reply:
Ah ok, I'll take this question out of this PR :)